faster_sam/lambda_event.py

```diff
@@ -141,12 +141,18 @@ async def event_builder(self):
         json_body = bytes_body.decode()
         body = json.loads(json_body)
 
-        attributes = {
-            "ApproximateReceiveCount": body["deliveryAttempt"],
-            "SentTimestamp": datetime.timestamp(
+        try:
+            timestamp = datetime.timestamp(
                 datetime.strptime(body["message"]["publishTime"], "%Y-%m-%dT%H:%M:%S.%fZ")
             )
-            * KILO_SECONDS,
+        except ValueError:
+            timestamp = datetime.timestamp(
+                datetime.strptime(body["message"]["publishTime"], "%Y-%m-%dT%H:%M:%S%z")
```

Comment:
O `Z` no final do formato ficou com letra minúscula, não tenho certeza se isso não gera um erro.

Reply:
Estava dando uma estudadinha básica aqui Julies kkkk 🙃

[strftime() and strptime() Behavior](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior)
[Parse ISO 8601 date ending in Z with Python](https://blog.ham1.co.uk/2023/05/29/python-parse-iso-date-with-z-at-end/#:~:text=The%20'Z'%20at%20the%20end,can%20use%20the%20datetime%20module.)

O que eu entendi:

O `Z` maiúsculo, indica que a hora está no formato UTC, ele espera que a string termine com `Z`... Que é o formato que o pubsub vai enviar.

Exemplo -> `2024-07-23T15:30:45.123456Z`

Agora o `%z` minúsculo, é usado para lidar com deslocamento do fuso horário, que pode vir no formato `±HHMM` ou `±HHMMSS`.

Exemplo -> `2024-07-23T15:30:45-0400`

Então o certo nesse caso seria usar o formato assim: `%Y-%m-%dT%H:%M:%SZ`, acabo de corrigir...